---
title: "Untitled"
author: "Marina Alexander"
date: "02/09/2020"
output:
  pdf_document: default
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}

# remind R where to look for libraries
.libPaths(c("C:/Users/ale097/Data School/Packages"))
# load libraries
library(tidyverse)
library(dplyr)
library(knitr)
library(dplyr)
library(readr)
library(vroom)
library(stringr)
library(EnhancedVolcano)
library(viridis)
library(ggsci)
library(cowplot)
library(patchwork)
library(ggsignif)
library(DESeq2)
library(biomaRt)

knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)
```

```{r }

rm(list=ls())

SARSCO2_Calu3_counts <- read.csv("data/SARSCOV2_Calu3_HTSeq_merged_gtf_counts.csv")
SARSCO2_Calu3_metadata <- read.csv("data/metadata_mRNA_SARSCOV2_Calu3.csv")

df_merged <- read.csv("data/SARSCOV2_Calu3_HTSeq_merged_gtf_counts.csv", header=TRUE, row.names=1)

dim(df_merged)

md_merged <- read.csv("data/metadata_mRNA_SARSCOV2_Calu3_merged.csv", header=TRUE, row.names=1)

dim(md_merged)

#merge count and metadata file
all(rownames(md_merged) == colnames (df_merged))

# Calculate counts per million.
# Filter rows: at least 3 samples must have at least 1 cpm.
# Retain rows according to 'keep' and nothing on columns.
cpm <- apply(df_merged, 2, function(x) (x/sum(x))*1000000)
keep <- rowSums( cpm >= 1 ) >=4
df_filtered_merged <- df_merged[ keep, ]

dim(df_filtered_merged)


# Construct a SummarizedExperiment object:
dds_merged <- DESeqDataSetFromMatrix(
  countData = df_filtered_merged,
  colData = md_merged,
  design = ~ condition) # ~ is representative of 'by', i.e. compare by condition


# Perform DE testing:
dds_merged <- DESeq(dds_merged)

# Output normalized counts:
norm_counts_merged <- counts (dds_merged, normalized=TRUE)
write.csv(norm_counts_merged, file="results/DESeq2_SARSCOV2_Calu3_all_normalized_counts_merged.csv")


# Convert results to dataframe:
CoV2genes24hr_merged <- results(dds_merged, contrast = c("condition", "SARSCOV2_24hr", "uninfected_24hr"))

CoV2genes24hr_merged %>% 
  as_tibble(rownames = "gene")

```

```{r listing the significant DE genes }
# Set adjusted p-value significance (padj) threshold:
alpha <- c( 0.05 )
# Set log2FoldChange threshold:
# As it's log2, > 1 is actually equal to 2-fold change or above.
beta <- c( 1 )
# Set baseMean threshold:
gamma <- c( 5 )

# 'Which' provides positions of 'TRUE' values.
sigCoV2genes24hr_merged<- CoV2genes24hr_merged[ which( CoV2genes24hr_merged$padj < alpha), ]
# Slices out rows where the adjusted p-value is <0.05.
sigCoV2genes24hr_merged <- sigCoV2genes24hr_merged[ which( abs(sigCoV2genes24hr_merged$log2FoldChange) > beta), ]
# Slices out rows where the fold change is above 2 and below -2.
# Abs=absolute, tells it to filter things above 2, ignoring whether value is positive or negative.
sigCoV2genes24hr_merged <- sigCoV2genes24hr_merged[ which(sigCoV2genes24hr_merged$baseMean > gamma), ]
write.csv(sigCoV2genes24hr_merged, file="results/sigCoV2genes24hr_merged.csv")
# Slices out rows above an average count of 10 reads (anything below is rubbish).

CoV2genes24hr_hits_merged <- rownames(sigCoV2genes24hr_merged)
length(CoV2genes24hr_hits_merged)
write.csv(norm_counts_merged[CoV2genes24hr_hits_merged, ], file="results/DESeq2_sig_CoV2genes24hr_normalized_counts_merged.csv")

SARSCOV2_DEgenes_Calu3_24hrs_merged <- read_csv("results/sigCoV2genes24hr_merged.csv") %>%
  as.data.frame() %>% 
  dplyr::rename(GENE_LOCUS = X1) %>%
  arrange(padj)

write.csv(SARSCOV2_DEgenes_Calu3_24hrs_merged, "results/SARSCOV2_DEgenes_Calu3_24hrs_merged.csv")

SARSCOV2_DEgenes_Calu3_24hrs_david_merged <- SARSCOV2_DEgenes_Calu3_24hrs_merged %>% 
  dplyr::select(gene)

write.csv(SARSCOV2_DEgenes_Calu3_24hrs_david_merged, "results/SARSCOV2_DEgenes_Calu3_24hrs_david_merged.csv")

```


```{r PCA  }

#join with the counts data 

all_expression <- read_csv("results/DESeq2_SARSCOV2_Calu3_all_normalized_counts.csv") %>%
  gather(name, norm_counts, -X1) %>%
  left_join(SARSCO2_Calu3_metadata, by = "name")




#scale gene expression for all samples 
scaled_genes <- all_expression %>%
  spread(X1, norm_counts) %>%
  select( -cells, -infection, -rep, -condition, -timepoint) %>% 
  column_to_rownames("name") %>% 
  scale()


#use the prcomp function on scaled samples
pca_genes <- prcomp(scaled_genes)

#tidy data frame for plotting
PCA_data <- pca_genes$x %>% 
  as_tibble(rownames = "name") %>%
  gather(PC, variance, -name) %>% 
  left_join(SARSCO2_Calu3_metadata, by = "name") %>%
  spread(PC, variance) %>% 
  separate(name, c("extraction", "sample", "mapping_index"))

#plot to examine variance for all samples 
pca_plot_sample <- ggplot(PCA_data, aes(x = PC1, y = PC2)) +
  geom_text(aes(label = sample), size = 4)
  
ggsave(filename = "results/pca_plot_samples.png", plot = pca_plot_sample, width = 12, height = 10, dpi = 300, units = "cm")


```

```{r looing at ACE2 in Calu3 cells}

ACE2_expression <- read_csv("results/DESeq2_SARSCOV2_Calu3_all_normalized_counts.csv") %>%
  gather(name, norm_counts, -X1) %>%
  left_join(SARSCO2_Calu3_metadata, by = "name") %>% 
  separate(X1, c("gene", "gene1")) %>%
  filter(gene == "ACE2")

plot_ACE2_Calu3 <- ggplot(ACE2_expression, aes(y = norm_counts, x = condition))+
  geom_jitter(width = 0.2)+
  labs(title = "ACE2 expression in Calu3 cells")

ggsave(filename = "results/plot_ACE2_Calu3.png", plot = plot_ACE2_Calu3, width = 12, height = 10, dpi = 300, units = "cm")


```

```{r Preparing count table for RiboToolkit}
#############################

library(org.Hs.eg.db)

ensemblsIDS <- read.csv("data/hg38.geneID.csv")

symbols <- mapIds(org.Hs.eg.db, keys = ensemblsIDS$GeneID, keytype = "ENSEMBL", column="SYMBOL") %>% 
  as.data.frame() %>%
  tibble::rownames_to_column("ensemblGeneIDs") %>% 
  dplyr::rename("GeneID" = ".")

HTseq_counts_org <- read.csv("data/SARSCOV2_Calu3_HTSeq_merged_gtf_counts.csv") %>% 
  separate(Gene.ID.Name, c("XLOC", "XLOC.ID", "GeneID")) %>%
  left_join(symbols, by ="GeneID") %>%
  dplyr::filter(ensemblGeneIDs != "NA") %>% 
  dplyr::select(ensemblGeneIDs, mRNA_01_merged,mRNA_01_merged, mRNA_02_merged, mRNA_03_merged, mRNA_04_merged, mRNA_05_merged, mRNA_06_merged, mRNA_07_merged, mRNA_08_merged, mRNA_09_merged )

############################

library(EnsDb.Hsapiens.v79)

#Convert from ensembl.gene to gene.symbol

geneIDs1 <- ensembldb::select(EnsDb.Hsapiens.v79, keys= ensemblsIDS$GeneID, keytype = "GENEID", columns = c("SYMBOL","GENEID"))

HTseq_counts_EnsDb <- read.csv("data/SARSCOV2_Calu3_HTSeq_merged_gtf_counts.csv") %>% 
  separate(Gene.ID.Name, c("XLOC", "XLOC.ID", "SYMBOL")) %>%
  left_join(geneIDs1, by = "SYMBOL") %>%
  dplyr::filter(GENEID != "NA") %>% 
  dplyr::select(GENEID, mRNA_01_merged,mRNA_01_merged, mRNA_02_merged, mRNA_03_merged, mRNA_04_merged, mRNA_05_merged, mRNA_06_merged, mRNA_07_merged, mRNA_08_merged, mRNA_09_merged )

count_RiboToolkit <- HTseq_counts_EnsDb %>% 
  dplyr::rename(Geneid = GENEID, sample1 = mRNA_01_merged, sample2 = mRNA_02_merged, sample3 = mRNA_03_merged, sample4 = mRNA_04_merged, sample5 = mRNA_05_merged, sample6 = mRNA_06_merged, sample7 = mRNA_07_merged, sample8 = mRNA_08_merged, sample9 = mRNA_09_merged) %>% 
  dplyr::select(Geneid, sample1, sample2, sample3, sample7, sample8, sample9)
  
  
write.table(count_RiboToolkit, "results/HTSeqcounts.txt", row.names = FALSE, col.names = TRUE)

read.table("data/htseq-countNCBI/mRNA_01_NCBI.count.txt")

######################################################

```


```{r Volcano plots  }


forplotting_ILRUNgenesUninf6hr <- ILRUNgenesUninf6hr %>%
  as.data.frame() %>% 
  rownames_to_column() %>% 
  separate(rowname, c("gene", "gene1"), sep = "_") %>% 
  select(-gene1) %>% 
  mutate(gene = str_replace(gene, "C6orf106", "ILRUN")) %>% 
  mutate(gene = str_replace(gene, "ACKR3", "CXCR7")) %>%
  mutate(gene = str_replace(gene, "CHUK","IkBKA")) %>%
  mutate(gene = str_replace(gene, "TNFSF10", "TRAIL")) %>% 
  remove_rownames %>% 
  column_to_rownames(var="gene")

write.csv(forplotting_ILRUNgenesUninf6hr, "results/ILRUNgenesUninf6hr_forplotting.csv")

##show only significant immune genes ###

APC <- read.csv(("data/KEGG_Immune.system_Antigen.processing.and.presentation_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

B_cell <- read.csv(("data/KEGG_Immune.system_B.cell.receptor.signalling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Chemokine <- read.csv(("data/KEGG_Immune.system_Chemokine.signaling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Complement <- read.csv(("data/KEGG_Immune.system_Complement.and.coagulation.cascades_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Lectin <- read.csv(("data/KEGG_Immune.system_C-type.lectin.receptor.signalling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

DNAsensing <- read.csv(("data/KEGG_Immune.system_Cytosolic.DNA-sensing.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Fcepsilon <- read.csv(("data/KEGG_Immune.system_Fc.epsilon.RI.signaling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Fcgamma <- read.csv(("data/KEGG_Immune.system_Fc.gamma.R-mediated.phagocytosis_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Hematopoetic <- read.csv(("data/KEGG_Immune.system_Hematopoietic.cell.lineage_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

IL17 <- read.csv(("data/KEGG_Immune.system_IL-17.signaling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

IntestinalIgA <- read.csv(("data/KEGG_Immune.system_Intestinal.immune.network.for.IgA.production_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

LeukocyeMigration <- read.csv(("data/KEGG_Immune.system_Leukocyte.transendothelial.migration_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

NKT <- read.csv(("data/KEGG_Immune.system_Natural.killer.cell.mediated.cytotoxicity_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

NOD <- read.csv(("data/KEGG_Immune.system_NOD-like.receptor.signaling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Platelet <- read.csv(("data/KEGG_Immune.system_Platelet.activation_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

RIGI <- read.csv(("data/KEGG_Immune.system_RIG-I-like.receptor.signaling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Tcell <- read.csv(("data/KEGG_Immune.system_T.cell.receptor.signalling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Th1Th2 <- read.csv(("data/KEGG_Immune.system_Th1.and.Th2.cell.differentiation_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene) %>% 
  filter(gene != "	 M ")

Th17 <- read.csv(("data/KEGG_Immune.system_Th17.cell.differentiation_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

Toll <- read.csv(("data/KEGG_Immune.system_Toll-like.receptor.signaling.pathway_genes.csv"),  row.names=NULL, header = FALSE) %>%
  as_tibble() %>% 
  separate(V1, c("desription", "gene"), sep = "RefSeq") %>% 
  mutate(gene_id = str_remove_all(gene, "[\\),]")) %>% 
  select(gene_id) %>% 
  filter(gene_id != "NA") %>% 
  separate(gene_id, c("gene", "extra"), sep = ";") %>%
  select(gene)

KEGG_immune_system_joined <- full_join(APC, B_cell, by = "gene") %>% 
  full_join(Chemokine,by = "gene") %>% 
  full_join(Complement,by = "gene") %>% 
  full_join(DNAsensing,by = "gene") %>% 
  full_join(Fcepsilon,by = "gene") %>% 
  full_join(Fcgamma,by = "gene") %>% 
  full_join(Hematopoetic,by = "gene") %>%
  full_join(IL17,by = "gene") %>% 
  full_join(IntestinalIgA,by = "gene") %>% 
  full_join(Lectin,by = "gene") %>% 
  full_join(LeukocyeMigration, by = "gene") %>% 
  full_join(NKT,by = "gene") %>% 
  full_join(NOD,by = "gene") %>% 
  full_join(Platelet,by = "gene") %>% 
  full_join(RIGI,by = "gene") %>% 
  full_join(Th17,by = "gene") %>% 
  full_join(Tcell,by = "gene") %>% 
  full_join(Th1Th2,by = "gene") %>% 
  full_join(Toll,by = "gene")

#### Trim all the white space ###########
KEGG_immune_system <- stri_trim(KEGG_immune_system_joined$gene) %>% 
  as_tibble() %>%
  rename(gene = value) %>% 
  filter(gene != "M")

tibble_ILRUNgenesUninf6hr <- ILRUNgenesUninf6hr %>%
  as.data.frame() %>% 
  rownames_to_column() %>% 
  separate(rowname, c("gene", "gene1"), sep = "_") %>% 
  select(-gene1)

#526 genes of the immune system detected
#344 genes significantly different between siNEG and siILRUN 
#57 genes with log2FC more than 0.75

immune_genes_Uninf6hr <- KEGG_immune_system %>%
  left_join(tibble_ILRUNgenesUninf6hr , by = "gene") %>%
  filter(baseMean != "NA") %>% 
  filter(padj <0.05 ) %>% 
  filter(log2FoldChange >0.75 | log2FoldChange < -0.75) 
  
mutate(gene = str_replace(gene, "C6orf106", "ILRUN")) %>% 
  mutate(gene = str_replace(gene, "ACKR3", "CXCR7")) %>%
  mutate(gene = str_replace(gene, "CHUK","IkBKA")) %>%
  mutate(gene = str_replace(gene, "TNFSF10", "TRAIL"))



#########################################

volcano_plot_ILRUNgenesUninf6hr <- EnhancedVolcano(forplotting_ILRUNgenesUninf6hr,
                                                 lab = rownames(forplotting_ILRUNgenesUninf6hr),
                                                 x = 'log2FoldChange',
                                                 y = 'padj',
                                                 selectLab = c(immune_genes_Uninf6hr$gene),
                                                 xlim = c(-3, 3),
                                                 title = 'ILRUN genes 6hr',
                                                 pCutoff = 0.05,
                                                 FCcutoff = 0.5,
                                                 pointSize = 4.0,
                                                 labSize = 5.0,
                                                 labCol = 'black',
                                                 labFace = 'bold',
                                                 boxedLabels = TRUE,
                                                 colAlpha = 4/5,
                                                 legendPosition = 'right',
                                                 legendLabSize = 14,
                                                 legendIconSize = 4.0,
                                                 drawConnectors = TRUE,
                                                 widthConnectors = 1.0,
                                                 colConnectors = 'black')


ggsave(filename = "results/volcano_plot_ILRUNgenesUninf6hr.png", plot = volcano_plot_ILRUNgenesUninf6hr, width = 30, height = 25, dpi = 300, units = "cm")


```

