---
title: "CDS_mapped_reads"
author: "Marina Alexander"
date: "08/10/2020"
output:
  pdf_document: default
  html_document:
    code_folding: hide
---

```{r setup, include=FALSE}

# remind R where to look for libraries
.libPaths(c("C:/Users/ale097/Data School/Packages"))
# load libraries
library(tidyverse)
library(dplyr)
library(knitr)
library(readr)
library(vroom)
library(stringr)
library(EnhancedVolcano)
library(viridis)
library(ggsci)
library(cowplot)
library(patchwork)
library(ggsignif)
library(DESeq2)
library(biomaRt)

knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = 'center'
)
```

```{r}

CDS_mapped <- read.csv("data/read_lengths_CDS.csv") %>%
  gather(sample, counts, -read_length)

plot01 <- CDS_mapped %>% 
  filter(sample == "sample_01") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample01")
plot02 <- CDS_mapped %>% 
  filter(sample == "sample_02") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample02")
plot03 <- CDS_mapped %>% 
  filter(sample == "sample_03") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample03")
plot04 <- CDS_mapped %>% 
  filter(sample == "sample_04") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample04")
plot05 <- CDS_mapped %>% 
  filter(sample == "sample_05") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample05")
plot06 <- CDS_mapped %>% 
  filter(sample == "sample_06") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample06")
plot07 <- CDS_mapped %>% 
  filter(sample == "sample_07") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample07")
plot08 <- CDS_mapped %>% 
  filter(sample == "sample_08") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample08")
plot09 <- CDS_mapped %>% 
  filter(sample == "sample_09") %>% 
  ggplot(aes(x = read_length, y = counts)) +
  geom_bar(stat="identity")+
  labs(title = "sample09")

plot_all <- plot_grid(plot01, plot02, plot03, plot04, plot05, plot06, plot07, plot08, plot09)

ggsave(filename = "results/allsamples_CDSreadlengths.png", plot = plot_all, width = 15, height = 10, dpi = 300, units = "cm")

```

```{r}

CDS_mapped_freq <- read.csv("data/read_lengths_CDS_frequency.csv") %>%
  gather(sample, frequency, -read_length)

write.csv(CDS_mapped_freq, "results/gathered_CDS_mapped_freq.csv")

CDS_mapped_freq_timepoint <- read.csv("results/gathered_CDS_mapped_freq_timepoint.csv")

CDS_mapped_freq_timepoint$timepoint <- factor(CDS_mapped_freq_timepoint$timepoint, levels=c("0hr", "6hr", "24hr"))

freq_readlength_plot <- ggplot(CDS_mapped_freq_timepoint, aes(x = read_length, y = frequency, color = sample))+
  geom_line()+
  geom_vline(xintercept = 28, color = "red", linetype = "dashed")+
  facet_wrap(~timepoint, nrow = 3)+
  ylim(0, 18)+
  xlim(15, 41)+
  labs(y = "frequency (%)",
       x = "read length", 
       title = "Read length distribution for footprints mapping to CDS" )

ggsave(filename = "results/freq_readlength_plot.png", plot = freq_readlength_plot, width = 20, height = 20, dpi = 300, units = "cm")

```

```{r}
library(readr)

sample1_total_reads <- read.csv("data/sample1_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s1")
sample1_RPKM_stats <- read.csv("data/sample1_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host, Chr) %>%
  mutate(sample = "s1") %>%
  mutate(timepoint = "0hr") %>%
  mutate(RPM = (CDS/sample1_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>% 
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM)) 

sample2_total_reads <- read.csv("data/sample2_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s2")
sample2_RPKM_stats <- read.csv("data/sample2_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host) %>%
  mutate(sample = "s2") %>%
  mutate(timepoint = "0hr") %>%
  mutate(RPM = (CDS/sample2_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>% 
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM)) 

sample3_total_reads <- read.csv("data/sample3_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s3")
sample3_RPKM_stats <- read.csv("data/sample3_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host) %>%
  mutate(sample = "s3") %>%
  mutate(timepoint = "0hr") %>%
  mutate(RPM = (CDS/sample3_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>% 
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM)) 

sample4_total_reads <- read.csv("data/sample4_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s4")
sample4_RPKM_stats <- read.csv("data/sample4_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host) %>%
  mutate(sample = "s4") %>%
  mutate(timepoint = "6hr") %>%
  mutate(RPM = (CDS/sample4_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>% 
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM)) 

sample5_total_reads <- read.csv("data/sample5_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s5")
sample5_RPKM_stats <- read.csv("data/sample5_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host) %>%
  mutate(sample = "s5") %>%
  mutate(timepoint = "6hr") %>%
  mutate(RPM = (CDS/sample5_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>%
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM)) 

sample6_total_reads <- read.csv("data/sample6_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s6")
sample6_RPKM_stats <- read.csv("data/sample6_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host) %>%
  mutate(sample = "s6") %>%
  mutate(timepoint = "6hr") %>%
  mutate(RPM = (CDS/sample6_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>%
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM)) 


sample7_total_reads <- read.csv("data/sample7_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s7")
sample7_RPKM_stats <- read.csv("data/sample7_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host) %>%
  mutate(sample = "s7") %>%
  mutate(timepoint = "24hr") %>%
  mutate(RPM = (CDS/sample7_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>%
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM)) 


sample8_total_reads <- read.csv("data/sample8_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s8")
sample8_RPKM_stats <- read.csv("data/sample8_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host) %>%
  mutate(sample = "s8") %>%
  mutate(timepoint = "24hr") %>%
  mutate(RPM = (CDS/sample8_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>%
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM))

sample9_total_reads <- read.csv("data/sample9_CDS.csv", skip =1) %>% 
  summarise(total_reads = sum(CDS)) %>% 
  mutate(sample= "s9")
sample9_RPKM_stats <- read.csv("data/sample9_CDS.csv", skip = 1) %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  dplyr::select(Geneid, Length, CDS, host) %>%
  mutate(sample = "s9") %>%
  mutate(timepoint = "24hr") %>%
  mutate(RPM = (CDS/sample9_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  filter(RPKM >100 ) %>%
  group_by(host, sample, timepoint) %>% 
  summarise(RPKM_mean_per_gene = mean(RPKM)) 

all_samples_RF_libsize <- bind_rows(sample1_total_reads, sample2_total_reads, sample3_total_reads, sample4_total_reads, sample5_total_reads, sample6_total_reads, sample7_total_reads, sample8_total_reads, sample9_total_reads) %>% 
  mutate(sample = str_replace_all(sample, c("s1" = "RF_CDS_s1", "s2" = "RF_CDS_s2", "s3" = "RF_CDS_s3", "s4" = "RF_CDS_s4", "s5" = "RF_CDS_s5",
                                            "s6" = "RF_CDS_s6", "s7" = "RF_CDS_s7", "s8" = "RF_CDS_s8", "s9" = "RF_CDS_s9")))


                                       
all_samples <- bind_rows(sample2_RPKM_stats, sample3_RPKM_stats, sample4_RPKM_stats, sample5_RPKM_stats, sample6_RPKM_stats, sample7_RPKM_stats, sample8_RPKM_stats, sample9_RPKM_stats) %>% 
  mutate(genome = str_replace_all(host, c("FALSE" = "SARSCOV2", "TRUE" = "human")))
  
all_samples$timepoint <- factor(all_samples$timepoint, levels = c("0hr", "6hr", "24hr"))  

plot_RPKM <- ggplot(all_samples, aes(y= RPKM_mean_per_gene, x= timepoint, color = genome))+
  geom_boxplot()+
  scale_y_log10()


```

```{r making a counts table with mRNA and footprints}

######### mRNA counts table #####################

mRNA_genes <- read.csv("data/mRNA_CDS_01.csv", skip =1) %>% 
  dplyr::select(Geneid, Chr, Length)

mRNA_sample1 <- read.csv("data/mRNA_CDS_01.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s1)
mRNA_sample2 <- read.csv("data/mRNA_CDS_02.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s2)
mRNA_sample3 <- read.csv("data/mRNA_CDS_03.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s3)
mRNA_sample4 <- read.csv("data/mRNA_CDS_04.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s4)
mRNA_sample5 <- read.csv("data/mRNA_CDS_05.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s5)
mRNA_sample6 <- read.csv("data/mRNA_CDS_06.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s6)
mRNA_sample7 <- read.csv("data/mRNA_CDS_07.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s7)
mRNA_sample8 <- read.csv("data/mRNA_CDS_08.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s8)
mRNA_sample9 <- read.csv("data/mRNA_CDS_09.csv", skip =1) %>% 
  dplyr::select(mRNA_CDS_s9)

##### total read counts ############
sample1_total_reads_mRNA <- read.csv("data/mRNA_CDS_01.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s1)) %>%
   mutate(sample= "mRNA_CDS_s1")
sample2_total_reads_mRNA <- read.csv("data/mRNA_CDS_02.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s2)) %>% 
  mutate(sample= "mRNA_CDS_s2")
sample3_total_reads_mRNA <- read.csv("data/mRNA_CDS_03.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s3)) %>% 
  mutate(sample= "mRNA_CDS_s3")
sample4_total_reads_mRNA <- read.csv("data/mRNA_CDS_04.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s4)) %>% 
  mutate(sample= "mRNA_CDS_s4")
sample5_total_reads_mRNA <- read.csv("data/mRNA_CDS_05.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s5)) %>% 
  mutate(sample= "mRNA_CDS_s5")
sample6_total_reads_mRNA <- read.csv("data/mRNA_CDS_06.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s6)) %>% 
  mutate(sample= "mRNA_CDS_s6")
sample7_total_reads_mRNA <- read.csv("data/mRNA_CDS_07.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s7)) %>% 
  mutate(sample= "mRNA_CDS_s7")
sample8_total_reads_mRNA <- read.csv("data/mRNA_CDS_08.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s8)) %>% 
  mutate(sample= "mRNA_CDS_s8")
sample9_total_reads_mRNA <- read.csv("data/mRNA_CDS_09.csv", skip =1) %>% 
  summarise(total_reads = sum(mRNA_CDS_s9)) %>% 
  mutate(sample= "mRNA_CDS_s9")

all_samples_mRNA_libsize <- bind_rows(sample1_total_reads_mRNA, sample2_total_reads_mRNA, sample3_total_reads_mRNA, sample4_total_reads_mRNA,
                                      sample5_total_reads_mRNA,sample6_total_reads_mRNA, sample7_total_reads_mRNA, sample8_total_reads_mRNA,
                                      sample9_total_reads_mRNA)

mRNA_CDS_counts <- bind_cols(mRNA_genes,mRNA_sample1, mRNA_sample2, mRNA_sample3, mRNA_sample4, mRNA_sample5, mRNA_sample6, mRNA_sample7,
                             mRNA_sample8, mRNA_sample9)

write.csv(mRNA_CDS_counts, "data/mRNA_CDS_counts.csv")

######### same for ribosome footprints #####################


RF_genes <- read.csv("data/sample1_CDS.csv", skip =1) %>% 
  dplyr::select(Geneid, Chr, Length)

RF_sample1 <- read.csv("data/sample1_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s1= CDS) %>% 
  dplyr::select(RF_CDS_s1)
RF_sample2 <- read.csv("data/sample2_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s2= CDS) %>% 
  dplyr::select(RF_CDS_s2)
RF_sample3 <- read.csv("data/sample3_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s3= CDS) %>% 
  dplyr::select(RF_CDS_s3)
RF_sample4 <- read.csv("data/sample4_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s4= CDS) %>% 
  dplyr::select(RF_CDS_s4)
RF_sample5 <- read.csv("data/sample5_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s5= CDS) %>% 
  dplyr::select(RF_CDS_s5)
RF_sample6 <- read.csv("data/sample6_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s6= CDS) %>% 
  dplyr::select(RF_CDS_s6)
RF_sample7 <- read.csv("data/sample7_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s7= CDS) %>% 
  dplyr::select(RF_CDS_s7)
RF_sample8 <- read.csv("data/sample8_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s8= CDS) %>% 
  dplyr::select(RF_CDS_s8)
RF_sample9 <- read.csv("data/sample9_CDS.csv", skip =1) %>% 
  dplyr::rename(RF_CDS_s9= CDS) %>% 
  dplyr::select(RF_CDS_s9)

RF_CDS_counts <- bind_cols(RF_genes,RF_sample1, RF_sample2, RF_sample3, RF_sample4, RF_sample5, RF_sample6, RF_sample7,
                             RF_sample8, RF_sample9)

write.csv(RF_CDS_counts, "data/RF_CDS_counts.csv")

RF_mRNA_raw_counts <- left_join(mRNA_CDS_counts,RF_CDS_counts, by = "Geneid" ) %>% 
  dplyr::select(-Chr.y,-Length.y ) %>% 
  dplyr::rename(Chr =Chr.x) %>% 
  dplyr::rename(Length =Length.x) %>% 
  filter(Geneid != "ORF1a")

write.csv(RF_mRNA_raw_counts, "data/RF_mRNA_raw_counts")

############### CPM normalization ##########

RF_gather_join_normalize_host <- RF_CDS_counts %>% 
  gather(sample, counts, -Geneid, - Chr, -Length) %>% 
  left_join(all_samples_RF_libsize, by = "sample") %>%
  mutate(cpm_RF = (counts/total_reads)*1000000) %>% 
  mutate(RPKM_RF = (cpm_RF/Length)*1000 ) %>%
  dplyr::select(Geneid, sample, RPKM_RF) %>% 
  spread(sample, RPKM_RF, drop = TRUE) %>% 
  filter(str_detect(Geneid, "ENSG"), RF_CDS_s1>5 , RF_CDS_s2>5 , RF_CDS_s3>5 , RF_CDS_s4>5 , RF_CDS_s5>5 , RF_CDS_s6>5) %>%
  left_join(mRNA_genes, by = "Geneid")

RF_gather_join_normalize_virus <- RF_CDS_counts %>% 
  gather(sample, counts, -Geneid, - Chr, -Length) %>% 
  left_join(all_samples_RF_libsize, by = "sample") %>%
  mutate(cpm_RF = (counts/total_reads)*1000000) %>% 
  mutate(RPKM_RF = (cpm_RF/Length)*1000 ) %>%
  dplyr::select(Geneid, sample, RPKM_RF) %>% 
  spread(sample, RPKM_RF, drop = TRUE) %>%
  left_join(mRNA_genes, by = "Geneid") %>%
  filter(str_detect(Chr, "NC")) %>%
  filter(Geneid != "ORF1a")
          
norm_RF_counts <- bind_rows(RF_gather_join_normalize_virus,RF_gather_join_normalize_host )

mRNA_gather_join_normalize <- mRNA_CDS_counts %>% 
  gather(sample, counts, -Geneid, - Chr, -Length) %>% 
  left_join(all_samples_mRNA_libsize, by = "sample") %>%
  mutate(cpm_mRNA = (counts/total_reads)*1000000) %>% 
  mutate(RPKM_mRNA = (cpm_mRNA/Length)*1000 ) %>%
  dplyr::select(Geneid, sample, RPKM_mRNA) %>% 
  spread(sample, RPKM_mRNA, drop = TRUE)

tail(mRNA_gather_join_normalize, 11)
  
TE_table <- left_join(norm_RF_counts,mRNA_gather_join_normalize, by = "Geneid") %>% 
  mutate(TE_s1 = RF_CDS_s1/mRNA_CDS_s1) %>% 
  mutate(TE_s2 = RF_CDS_s2/mRNA_CDS_s2) %>% 
  mutate(TE_s3 = RF_CDS_s3/mRNA_CDS_s3) %>% 
  mutate(TE_s4 = RF_CDS_s4/mRNA_CDS_s4) %>% 
  mutate(TE_s5 = RF_CDS_s5/mRNA_CDS_s5) %>% 
  mutate(TE_s6 = RF_CDS_s6/mRNA_CDS_s6) %>% 
  mutate(TE_s7 = RF_CDS_s7/mRNA_CDS_s7) %>% 
  mutate(TE_s8 = RF_CDS_s8/mRNA_CDS_s8) %>% 
  mutate(TE_s9 = RF_CDS_s9/mRNA_CDS_s9) %>% 
  dplyr::select(Geneid,TE_s1, TE_s2, TE_s3, TE_s4, TE_s5, TE_s6, TE_s7, TE_s8, TE_s9) %>% 
  left_join(mRNA_genes, by = "Geneid")

#### looking at mitochondrial transcripts 
TE_tabl

```

```{r  using mitochondially encoded genes to act as normalisation control }

#Building a counts table for DESeq2
Geneid <- read.csv("data/sample1_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM") %>% 
  dplyr::select(Geneid)

sample1_select <- read.csv("data/sample1_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM") %>%
  mutate(RPM = (CDS/sample1_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s1 = RPKM) %>% 
  dplyr::select(RF_CDS_s1) 
sample2_select <- read.csv("data/sample2_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM")%>%
  mutate(RPM = (CDS/sample2_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s2 = RPKM) %>% 
  dplyr::select(RF_CDS_s2)
sample3_select <- read.csv("data/sample3_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM")%>%
  mutate(RPM = (CDS/sample3_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s3 = RPKM) %>% 
  dplyr::select(RF_CDS_s3)
sample4_select <- read.csv("data/sample4_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM") %>%
  mutate(RPM = (CDS/sample4_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s4 = RPKM) %>% 
  dplyr::select(RF_CDS_s4)
sample5_select <- read.csv("data/sample5_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM") %>%
  mutate(RPM = (CDS/sample5_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s5 = RPKM) %>% 
  dplyr::select(RF_CDS_s5)
sample6_select <- read.csv("data/sample6_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM") %>%
  mutate(RPM = (CDS/sample6_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s6 = RPKM) %>% 
  dplyr::select(RF_CDS_s6)
sample7_select <- read.csv("data/sample7_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM") %>%
  mutate(RPM = (CDS/sample7_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s7 = RPKM) %>% 
  dplyr::select(RF_CDS_s7)
sample8_select <- read.csv("data/sample8_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM")%>%
  mutate(RPM = (CDS/sample8_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s8 = RPKM) %>% 
  dplyr::select(RF_CDS_s8)
sample9_select <- read.csv("data/sample9_CDS.csv", skip =1) %>% 
  filter(Chr == "chrM") %>%
  mutate(RPM = (CDS/sample9_total_reads$total_reads)*1000000) %>% 
  mutate(RPKM = (RPM/Length)*1000) %>%
  dplyr::rename(RF_CDS_s9 = RPKM) %>% 
  dplyr::select(RF_CDS_s9)

all_samples_chrM <- bind_cols(Geneid,sample1_select,sample2_select,sample3_select,sample4_select,sample5_select,sample6_select,
                              sample7_select,sample8_select,sample9_select)

write.csv(all_samples_chrM, "data/chrM_normalized_CDS_counts.csv")

```

```{r}

#join with the counts data 

metadata <- read.csv("data/SARSCOV2_riboseq_metadata.csv")

mito_expression <- read_csv("data/chrM_normalized_CDS_counts.csv") %>%
  dplyr::select(-X1) %>% 
  gather(sample, norm_counts, -Geneid) %>%
  left_join(metadata, by = "sample")

#scale gene expression for all samples 
scaled_genes <- mito_expression %>%
  spread(Geneid, norm_counts) %>%
  dplyr::select(-infection, -condition, -timepoint) %>%
  column_to_rownames("sample") %>% 
  scale()


#use the prcomp function on scaled samples
pca_genes <- prcomp(scaled_genes)

#tidy data frame for plotting
PCA_data <- pca_genes$x %>% 
  as_tibble(rownames = "sample") %>%
  gather(PC, variance, -sample) %>% 
  left_join(metadata, by = "sample") %>%
  spread(PC, variance)

#plot to examine variance for all samples 
pca_plot_sample <- ggplot(PCA_data, aes(x = PC1, y = PC2, color = infection)) +
  geom_text(aes(label = sample), size = 4)+
  labs(title = "PCA on normalized mitochondrial CDS counts")

mito_expression_plot <- ggplot(mito_expression, aes(y = Geneid , x= norm_counts, color = infection))+
  geom_jitter(width = 0.1)+
  labs(title = "Mitochondrial translation")
  
ggsave(filename = "results/pca_plot_mito_genes.png", plot = pca_plot_sample, width = 12, height = 10, dpi = 300, units = "cm")


```

```{r   GeTMM using TMM normalization on length corrected transcripts for mito genes}

RF_counts <- read.csv("data/RF_CDS_counts.csv")
mRNA_counts <- read.csv("data/mRNA_CDS_counts.csv")

################# Ribosome fragments ##################################
#subset the mitogenes
mito_RF_counts <- RF_counts %>%
  dplyr::select(-X) %>% 
  filter(Chr == "chrM") %>% 
  mutate(RF_K_s1 = (RF_CDS_s1/Length)*1000) %>% 
  mutate(RF_K_s2 = (RF_CDS_s2/Length)*1000) %>% 
  mutate(RF_K_s3 = (RF_CDS_s3/Length)*1000) %>% 
  mutate(RF_K_s4 = (RF_CDS_s4/Length)*1000) %>% 
  mutate(RF_K_s5 = (RF_CDS_s5/Length)*1000) %>%
  mutate(RF_K_s6 = (RF_CDS_s6/Length)*1000) %>% 
  mutate(RF_K_s7 = (RF_CDS_s7/Length)*1000) %>% 
  mutate(RF_K_s8 = (RF_CDS_s8/Length)*1000) %>% 
  mutate(RF_K_s9 = (RF_CDS_s9/Length)*1000) %>% 
  dplyr::select(Geneid,RF_K_s1, RF_K_s2, RF_K_s3,RF_K_s4, RF_K_s5, RF_K_s6,RF_K_s7, RF_K_s8,RF_K_s9) %>% 
  column_to_rownames("Geneid")

library(edgeR)

#GeTMM
# for normalization purposes, no grouping of samples
group_RF <- c(rep("A",ncol(mito_RF_counts)))
mito_RF_norm <- DGEList(counts=mito_RF_counts,group=group_RF)
mito_RF_norm <- calcNormFactors(mito_RF_norm, method = "TMM")
mito_RF_norm_edgeR <- cpm(mito_RF_norm)

get_norm_factors <- mito_RF_norm$samples %>% 
  as_tibble(rownames = "sample") %>% 
  dplyr::select(norm.factors)
  

plot_boxplot_norm_RF <- mito_RF_norm_edgeR %>% 
  as_tibble(rownames = "Geneid") %>%
  gather(sample, norm_RF_K, -Geneid) %>%
  ggplot(aes(x= sample, y = norm_RF_K))+
  geom_boxplot()+
  scale_y_log10()+
  labs(y = "Ribosome fragment normalized TPM")


group_RF <- c(rep("A",ncol(mito_RF_counts)))
mito_RF <- DGEList(counts=mito_RF_counts,group=group_RF)
mito_RF <- calcNormFactors(mito_RF, method = "none")
mito_RF_edgeR <- cpm(mito_RF)



plot_boxplot_RF <- mito_RF_edgeR %>% 
  as_tibble(rownames = "Geneid") %>%
  gather(sample, RF_K, -Geneid) %>%
  ggplot(aes(x= sample, y = RF_K))+
  geom_boxplot()+
  scale_y_log10()+
  labs(y = "Ribosome fragment TPM")

boxplot_normalize_RF <- plot_grid(plot_boxplot_RF,plot_boxplot_norm_RF, ncol = 1)

ggsave(filename = "results/boxplot_normalize_RF_TMM.png", plot = boxplot_normalize_RF, width = 20, height = 15, dpi = 600, units = "cm")

################## mRNA ####################################

mito_mRNA_counts <- mRNA_counts %>%
  dplyr::select(-X) %>% 
  filter(Chr == "chrM") %>% 
  mutate(mRNA_K_s1 = (mRNA_CDS_s1/Length)*1000) %>% 
  mutate(mRNA_K_s2 = (mRNA_CDS_s2/Length)*1000) %>% 
  mutate(mRNA_K_s3 = (mRNA_CDS_s3/Length)*1000) %>% 
  mutate(mRNA_K_s4 = (mRNA_CDS_s4/Length)*1000) %>% 
  mutate(mRNA_K_s5 = (mRNA_CDS_s5/Length)*1000) %>%
  mutate(mRNA_K_s6 = (mRNA_CDS_s6/Length)*1000) %>% 
  mutate(mRNA_K_s7 = (mRNA_CDS_s7/Length)*1000) %>% 
  mutate(mRNA_K_s8 = (mRNA_CDS_s8/Length)*1000) %>% 
  mutate(mRNA_K_s9 = (mRNA_CDS_s9/Length)*1000) %>%
  dplyr::select(Geneid,mRNA_K_s1, mRNA_K_s2, mRNA_K_s3, mRNA_K_s4, mRNA_K_s5, mRNA_K_s6,mRNA_K_s7, mRNA_K_s8,mRNA_K_s9) %>% 
  column_to_rownames("Geneid")

group_mRNA <- c(rep("A",ncol(mito_mRNA_counts)))
mito_mRNA_norm <- DGEList(counts=mito_mRNA_counts,group=group_mRNA)
mito_mRNA_norm <- calcNormFactors(mito_mRNA_norm, method = "TMM")
mito_mRNA_norm_edgeR <- cpm(mito_mRNA_norm)

get_norm_factors_mRNA <- mito_mRNA_norm$samples %>% 
  as_tibble(rownames = "sample") %>% 
  dplyr::select(norm.factors)

plot_boxplot_norm_mRNA <- mito_mRNA_norm_edgeR %>% 
  as_tibble(rownames = "Geneid") %>%
  gather(sample, norm_RF_K, -Geneid) %>%
  ggplot(aes(x= sample, y = norm_RF_K))+
  geom_boxplot()+
  scale_y_log10()+
  labs(y = "mRNA normalized TPM")

group_mRNA <- c(rep("A",ncol(mito_mRNA_counts)))
mito_mRNA <- DGEList(counts=mito_mRNA_counts,group=group_mRNA)
mito_mRNA <- calcNormFactors(mito_mRNA, method = "none")
mito_mRNA_edgeR <- cpm(mito_mRNA)

plot_boxplot_mRNA <- mito_mRNA_edgeR %>% 
  as_tibble(rownames = "Geneid") %>%
  gather(sample,mRNA_K, -Geneid) %>%
  ggplot(aes(x= sample, y = mRNA_K))+
  geom_boxplot()+
  scale_y_log10()+
  labs(y = "mRNA TPM")

boxplot_normalize_mRNA <- plot_grid(plot_boxplot_mRNA,plot_boxplot_norm_mRNA, ncol = 1)

ggsave(filename = "results/boxplot_normalize_mRNA_TMM.png", plot = boxplot_normalize_mRNA, width = 20, height = 15, dpi = 600, units = "cm")




```


```{r   GeTMM using TMM normalization on length corrected transcripts for all genes}

RF_counts <- read.csv("data/RF_CDS_counts.csv")
mRNA_counts <- read.csv("data/mRNA_CDS_counts.csv")

################# Ribosome fragments ##################################
#subset the mitogenes
all_RF_counts <- RF_counts %>%
  dplyr::select(-X) %>% 
  mutate(RF_K_s1 = (RF_CDS_s1/Length)*1000) %>% 
  mutate(RF_K_s2 = (RF_CDS_s2/Length)*1000) %>% 
  mutate(RF_K_s3 = (RF_CDS_s3/Length)*1000) %>% 
  mutate(RF_K_s4 = (RF_CDS_s4/Length)*1000) %>% 
  mutate(RF_K_s5 = (RF_CDS_s5/Length)*1000) %>%
  mutate(RF_K_s6 = (RF_CDS_s6/Length)*1000) %>% 
  mutate(RF_K_s7 = (RF_CDS_s7/Length)*1000) %>% 
  mutate(RF_K_s8 = (RF_CDS_s8/Length)*1000) %>% 
  mutate(RF_K_s9 = (RF_CDS_s9/Length)*1000) %>% 
  dplyr::select(Geneid,RF_K_s1, RF_K_s2, RF_K_s3,RF_K_s4, RF_K_s5, RF_K_s6,RF_K_s7, RF_K_s8,RF_K_s9) %>% 
  column_to_rownames("Geneid")

library(edgeR)

#GeTMM
# for normalization purposes, no grouping of samples
group_RF_all <- c(rep("A",ncol(all_RF_counts)))
all_RF_norm <- DGEList(counts=all_RF_counts,group=group_RF_all)
all_RF_norm <- calcNormFactors(all_RF_norm, method = "TMM")

get_norm_library_size_RF <- all_RF_norm$samples %>%
  as_tibble(rownames = "sample") %>% 
  dplyr::select(sample, lib.size) %>% 
  bind_cols(get_norm_factors) %>%
  mutate(norm_library_size = lib.size*norm.factors ) %>% 
  dplyr::select(sample, norm_library_size )
  
mRNA_genes <- read.csv("data/mRNA_CDS_01.csv", skip =1) %>% 
  dplyr::select(Geneid, Chr, Length)


TPM_RF <- all_RF_counts %>% 
  as_tibble(rownames = "Geneid") %>% 
  gather(sample, RF_K, -Geneid) %>%
  left_join(get_norm_library_size_RF, by = "sample") %>% 
  mutate(TPM_RF = (RF_K/norm_library_size)*1000000) %>% 
  dplyr::select(Geneid, sample,TPM_RF ) %>% 
  spread(sample,TPM_RF ) %>%
  left_join(mRNA_genes, by = "Geneid")
  

mRNA_genes <- read.csv("data/mRNA_CDS_01.csv", skip =1) %>% 
  dplyr::select(Geneid, Chr, Length)


################## mRNA ####################################

all_mRNA_counts <- mRNA_counts %>%
  dplyr::select(-X) %>%
  mutate(mRNA_K_s1 = (mRNA_CDS_s1/Length)*1000) %>% 
  mutate(mRNA_K_s2 = (mRNA_CDS_s2/Length)*1000) %>% 
  mutate(mRNA_K_s3 = (mRNA_CDS_s3/Length)*1000) %>% 
  mutate(mRNA_K_s4 = (mRNA_CDS_s4/Length)*1000) %>% 
  mutate(mRNA_K_s5 = (mRNA_CDS_s5/Length)*1000) %>%
  mutate(mRNA_K_s6 = (mRNA_CDS_s6/Length)*1000) %>% 
  mutate(mRNA_K_s7 = (mRNA_CDS_s7/Length)*1000) %>% 
  mutate(mRNA_K_s8 = (mRNA_CDS_s8/Length)*1000) %>% 
  mutate(mRNA_K_s9 = (mRNA_CDS_s9/Length)*1000) %>%
  dplyr::select(Geneid,mRNA_K_s1, mRNA_K_s2, mRNA_K_s3, mRNA_K_s4, mRNA_K_s5, mRNA_K_s6,mRNA_K_s7, mRNA_K_s8,mRNA_K_s9) %>% 
  column_to_rownames("Geneid")

group_mRNA_all <- c(rep("A",ncol(all_mRNA_counts)))
all_mRNA_norm <- DGEList(counts=all_mRNA_counts,group=group_mRNA_all)
all_mRNA_norm <- calcNormFactors(all_mRNA_norm, method = "TMM")

get_norm_library_size_mRNA <- all_mRNA_norm$samples %>%
  as_tibble(rownames = "sample") %>% 
  dplyr::select(sample, lib.size) %>% 
  bind_cols(get_norm_factors_mRNA) %>%
  mutate(norm_library_size = lib.size*norm.factors ) %>% 
  dplyr::select(sample, norm_library_size )
  
TPM_mRNA <- all_mRNA_counts %>% 
  as_tibble(rownames = "Geneid") %>% 
  gather(sample, mRNA_K, -Geneid) %>%
  left_join(get_norm_library_size_mRNA, by = "sample") %>% 
  mutate(TPM_mRNA = (mRNA_K/norm_library_size)*1000000) %>% 
  dplyr::select(Geneid, sample,TPM_mRNA ) %>% 
  spread(sample,TPM_mRNA )

write.csv(TPM_mRNA, "data/TPM_mRNA.csv")

```


```{r}

TE_table <- left_join(TPM_RF, TPM_mRNA, by ="Geneid" ) %>% 
  mutate(s1 = RF_K_s1/mRNA_K_s1) %>% 
  mutate(s2 = RF_K_s1/mRNA_K_s2) %>% 
  mutate(s3 = RF_K_s1/mRNA_K_s3) %>% 
  mutate(s4 = RF_K_s1/mRNA_K_s4) %>% 
  mutate(s5 = RF_K_s1/mRNA_K_s5) %>% 
  mutate(s6 = RF_K_s1/mRNA_K_s6) %>% 
  mutate(s7 = RF_K_s1/mRNA_K_s7) %>% 
  mutate(s8 = RF_K_s1/mRNA_K_s8) %>%
  mutate(s9 = RF_K_s1/mRNA_K_s9) %>% 
  dplyr::select(Geneid, s1, s2, s3, s4, s5, s6, s7, s8, s9)
  


```


```{r}
metadata <- read.csv("data/SARSCOV2_riboseq_metadata.csv")

immune_genes <- read.csv("data/InnateDB_genes.csv") %>% 
  filter(species == "Homo sapiens") %>% 
  dplyr::rename(Geneid = ensembl) %>% 
  dplyr::select(Geneid, name, function.)

TE_table_metadata <- read.csv("data/TE_table_correct.csv") %>% 
  gather(sample, TE, -Geneid, -host) %>% 
  left_join(metadata, by = "sample") %>%
  mutate(Geneid = gsub("\\.\\d+$","", Geneid)) %>%
  left_join(immune_genes) %>% 
  mutate(genome = ifelse(host == "TRUE", "host", "virus")) %>% 
  dplyr::select(-host) %>% 
  mutate(ontology = ifelse(name != "NA", "innate immune", "other")) %>% 
  filter(TE < 50)

TE_infected_6hr <- TE_table_metadata %>% 
  filter(timepoint == "6hr") %>% 
  dplyr::select(Geneid, TE, genome, ontology, name) %>% 
  dplyr::rename(TE_6hr = TE)

TE_uninfected <- TE_table_metadata %>% 
  filter(condition == "uninfected") %>% 
  dplyr::select(Geneid, TE, genome, ontology, name) %>% 
  dplyr::rename(TE_uninfected = TE)

combinded_TE <- left_join(TE_infected_6hr,TE_uninfected) %>% 
  filter(ontology == "innate immune")

TE_uninfected_above30 <- combinded_TE %>% 
  filter(TE_uninfected >25, TE_uninfected <50) %>% 
  filter(Geneid == "ENSG00000105810"| Geneid == "ENSG00000137801") %>%
  filter(TE_6hr <15, TE_6hr >11 )
  

plot_6hr_50 <- ggplot(combinded_TE, aes(x = TE_uninfected, y = TE_6hr))+
  geom_point(alpha = 0.15, size = 2)+
  ylim(0, 50)+
  geom_abline(color = "red", linetype = "dashed")+
  geom_text(data = TE_uninfected_above30, aes(x = TE_uninfected, y = TE_6hr, label = name), 
             hjust = 0.5, vjust = -0.5, size = 4, check_overlap=TRUE)
  
plot_6hr_10 <- ggplot(combinded_TE, aes(x = TE_uninfected, y = TE_6hr))+
  geom_point(alpha = 0.1, size = 2)+
  ylim(0, 10)+
  xlim(0, 10)+
  geom_abline(color = "red", linetype = "dashed")

ggsave(filename = "results/plot_6hr_scatter_50.png", plot = plot_6hr_50, width = 20, height = 20, dpi = 600, units = "cm")
ggsave(filename = "results/plot_6hr_scatter_10.png", plot = plot_6hr_10, width = 20, height = 20, dpi = 600, units = "cm")

```

```{r edgeR differential expression }

library(edgeR)

metadata_early <- metadata %>% 
  filter(condition != "infected_24hr") %>% 
  dplyr::select(sample, condition) %>%
  column_to_rownames("sample")
  
TE_table_early <- read.csv("data/TE_table_correct.csv") %>%
  filter(host != "FALSE") %>% 
  dplyr::select(Geneid, s1, s2, s3, s4, s5, s6) %>%
  filter(s1<50, s2<50, s3<50, s4<50, s5<50, s6<50)
  
TE_counts_early <- DGEList(counts = TE_table_early[, -1], genes = TE_table_early[, 1], samples = metadata_early)

TE_counts_early$genes

design_matrix_early <- model.matrix(~ condition, data = metadata_early)

modelling_TE_early <- voom(TE_counts_early, design = design_matrix_early)
vfit <- lmFit(modelling_TE_early, design_matrix_early)
efit <- eBayes(vfit)

TE_early_plot <- voom(TE_counts_early, plot = T, design = design_matrix_early)

coeff_TE <- topTable(efit, number = 8047) %>% 
  as.tibble() %>% 
  dplyr::select(genes, logFC, AveExpr, adj.P.Val) %>% 
  mutate(signif = -log10(adj.P.Val)) %>% 
  arrange(signif)


###################################################################################
immune_genes_volcano <- immune_genes %>% 
  dplyr::rename(genes = Geneid)

coeff_plotting <- topTable(efit, number = 8047) %>% 
  as.tibble() %>% 
  dplyr::select(genes, logFC, AveExpr, adj.P.Val) %>%
  mutate(genes = gsub("\\.\\d+$","", genes)) %>%
  mutate(signif = -log10(adj.P.Val)) %>%
  column_to_rownames("genes")

highlight <- coeff_TE %>%
  filter(adj.P.Val <= 0.01) %>% 
  filter(logFC >=1 | logFC <=-1) %>%
  arrange(logFC) %>% 
  mutate(genes = gsub("\\.\\d+$","", genes)) %>%
  mutate(signif = -log10(adj.P.Val)) %>%
  left_join(immune_genes_volcano, by = "genes") %>%
  filter(name != "NA")

volcano_plot_TE_early <- EnhancedVolcano(coeff_plotting,
    lab = rownames(coeff_plotting),
    selectLab = c(highlight$genes),
    labSize = 5,
    labFace = 'bold',
    drawConnectors = TRUE,
    x = "logFC",
    y = "adj.P.Val",
    xlim = c(-3.0, 3.0),
    ylim = c(0, 8),
    FCcutoff = 1,
    pCutoff = 0.01)

ggsave(filename = "results/volcano_plot_TE_early.png", plot = volcano_plot_TE_early, width = 30, height = 25, dpi = 300, units = "cm")


```

```{r  difefrential expression at the mRNA level}

mRNA_genes_Chr <- read.csv("data/mRNA_CDS_01.csv", skip =1) %>% 
  dplyr::select(Geneid, Chr)

metadata_early <- metadata %>% 
  filter(condition != "infected_24hr") %>% 
  dplyr::select(sample, condition) %>%
  column_to_rownames("sample")
  
mRNA_table_early <- read.csv("data/TPM_mRNA.csv") %>%
  dplyr::select(-X) %>%
  left_join(mRNA_genes_Chr, by = "Geneid") %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  filter(host != "FALSE") %>%
  dplyr::rename(s1 = mRNA_K_s1, s2 = mRNA_K_s2, s3 = mRNA_K_s3, s4 = mRNA_K_s4, s5 = mRNA_K_s5, s6 = mRNA_K_s6) %>% 
  dplyr::select(Geneid, s1, s2, s3, s4, s5, s6) %>% 
  filter(s1>5, s2 >5, s3 >5, s4 >5, s5 >5, s6 >5)
  
mRNA_counts_early <- DGEList(counts = mRNA_table_early[, -1], genes = mRNA_table_early[, 1], samples = metadata_early)

design_matrix_early <- model.matrix(~ condition, data = metadata_early)

modelling_mRNA_early <- voom(mRNA_counts_early, design = design_matrix_early)
vfit_mRNA <- lmFit(modelling_mRNA_early, design_matrix_early)
efit_mRNA <- eBayes(vfit_mRNA)

mRNA_early_plot <- voom(mRNA_counts_early, plot = T, design = design_matrix_early)

coeff_mRNA <- topTable(efit_mRNA, number = 6673) %>% 
  as.tibble() %>% 
  dplyr::select(genes, logFC, AveExpr, adj.P.Val) %>% 
  mutate(signif = -log10(adj.P.Val)) %>% 
  arrange(signif)

#################### Volcano plot ####################################
immune_genes_volcano <- immune_genes %>% 
  dplyr::rename(genes = Geneid)

coeff_plotting_mRNA <- topTable(efit_mRNA, number = 6673) %>% 
  as.tibble() %>% 
  dplyr::select(genes, logFC, AveExpr, adj.P.Val) %>%
  mutate(genes = gsub("\\.\\d+$","", genes)) %>%
  mutate(signif = -log10(adj.P.Val)) %>%
  column_to_rownames("genes")

highlight_mRNA <- coeff_mRNA %>%
  filter(adj.P.Val <= 0.01) %>% 
  filter(logFC >=1 | logFC <=-1) %>%
  arrange(logFC) %>% 
  mutate(genes = gsub("\\.\\d+$","", genes)) %>%
  mutate(signif = -log10(adj.P.Val)) %>%
  left_join(immune_genes_volcano, by = "genes") %>%
  filter(name != "NA")

volcano_plot_mRNA_early <- EnhancedVolcano(coeff_plotting_mRNA,
    lab = rownames(coeff_plotting_mRNA),
    selectLab = c(highlight_mRNA$genes),
    labSize = 5,
    labFace = 'bold',
    drawConnectors = TRUE,
    x = "logFC",
    y = "adj.P.Val",
    xlim = c(-3.0, 3.0),
    ylim = c(0, 10),
    FCcutoff = 1,
    pCutoff = 0.01)

ggsave(filename = "results/volcano_plot_mRNA_early.png", plot = volcano_plot_mRNA_early, width = 30, height = 25, dpi = 300, units = "cm")

#####################################################################

coeff_mRNA_rename <- coeff_mRNA %>%
  dplyr::rename(logFC.mRNA = logFC, adj.P.Val.mRNA = adj.P.Val) %>%
  dplyr::select(genes,logFC.mRNA, adj.P.Val.mRNA)

coeff_TE_rename <- coeff_TE %>%
  dplyr::rename(logFC.TE = logFC, adj.P.Val.TE = adj.P.Val) %>%
  dplyr::select(genes,logFC.TE, adj.P.Val.TE)
  

TE_versus_mRNA <- left_join(coeff_mRNA_rename, coeff_TE_rename, by = "genes" ) %>% 
  filter(logFC.TE != "NA")

plot_TE_versus_mRNA <- ggplot(TE_versus_mRNA, aes( x= logFC.mRNA, y = logFC.TE))+
  geom_point()+
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  labs(x= "LogFC mRNA-seq (infected/uninfected", 
       y = "LogFC Translation efficiency (infected/uninfected)")



```
```{r comparing two uninfected replicates }

#############  mRNA #####################################

mRNA_genes_Chr <- read.csv("data/mRNA_CDS_01.csv", skip =1) %>% 
  dplyr::select(Geneid, Chr)

metadata_uninfected <- read.csv("data/SARSCOV2_riboseq__rep_metadata.csv") %>% 
  filter(sample == "s1"| sample == "s2") %>% 
  dplyr::select(sample, rep ) %>%
  column_to_rownames("sample")
  
mRNA_table_uninfected <- read.csv("data/TPM_mRNA.csv") %>%
  dplyr::select(-X) %>%
  left_join(mRNA_genes_Chr, by = "Geneid") %>%
  mutate(host = case_when(str_detect(Geneid, "ENSG") ~ TRUE, TRUE ~ FALSE )) %>%
  filter(host != "FALSE") %>%
  dplyr::rename(s1 = mRNA_K_s1, s2 = mRNA_K_s2, s3 = mRNA_K_s3, s4 = mRNA_K_s4, s5 = mRNA_K_s5, s6 = mRNA_K_s6) %>% 
  dplyr::select(Geneid, s1, s2) %>% 
  filter(s1>5, s2 >5)
  
mRNA_counts_uninfected <- DGEList(counts = mRNA_table_uninfected[, -1], genes = mRNA_table_uninfected[, 1], samples = metadata_uninfected)

design_matrix_uninfected<- model.matrix(~ rep, data = metadata_uninfected)

modelling_mRNA_uninfected <- voom(mRNA_counts_uninfected, design = design_matrix_uninfected)
vfit_mRNA_uninfected <- lmFit(modelling_mRNA_uninfected, design_matrix_uninfected)
efit_mRNA_uninfected <- eBayes(vfit_mRNA_uninfected)

mRNA_uninfected_plot <- voom(mRNA_counts_uninfcted, plot = T, design = design_matrix_uninfected)

coeff_mRNA_uninfected <- topTable(efit_mRNA_uninfected, number = 6673) %>% 
  as.tibble() %>% 
  dplyr::select(genes, logFC, AveExpr, adj.P.Val) %>% 
  mutate(signif = -log10(adj.P.Val)) %>% 
  arrange(signif)

############# TE #####################################

  
TE_table_uninfected <- read.csv("data/TE_table_correct.csv") %>%
  filter(host != "FALSE") %>% 
  dplyr::select(Geneid, s1, s2) %>%
  filter(s1<50, s2<50)
  
TE_counts_uninfected <- DGEList(counts = TE_table_uninfcetd[, -1], genes = TE_table_uninfected[, 1], samples = metadata_uninfected)

design_matrix_uninfected <- model.matrix(~ condition, data = metadata_early)

modelling_TE_uninfected <- voom(TE_counts_uninfected, design = design_matrix_uninfected)
vfit_TE_uninfected <- lmFit(modelling_TE_early, design_matrix_uninfected)
efit_TE_uninfected <- eBayes(vfit_TE_uninfected)

TE_early_plot <- voom(TE_counts_uninfected, plot = T, design = design_matrix_uninfected)

coeff_TE_uninfected <- topTable(efit_TE_uninfected, number = 8047) %>% 
  as.tibble() %>% 
  dplyr::select(genes, logFC, AveExpr, adj.P.Val) %>% 
  mutate(signif = -log10(adj.P.Val)) %>% 
  arrange(signif)


##################  joining  TE and mRNA ##########################################

coeff_mRNA_rename_uninfected <- coeff_mRNA_uninfected %>%
  dplyr::rename(logFC.mRNA = logFC, adj.P.Val.mRNA = adj.P.Val) %>%
  dplyr::select(genes,logFC.mRNA, adj.P.Val.mRNA)

coeff_TE_rename_uninfected <- coeff_TE_uninfected %>%
  dplyr::rename(logFC.TE = logFC, adj.P.Val.TE = adj.P.Val) %>%
  dplyr::select(genes,logFC.TE, adj.P.Val.TE)
  

TE_versus_mRNA_uninfected <- left_join(coeff_mRNA_rename, coeff_TE_rename, by = "genes" ) %>% 
  filter(logFC.TE != "NA")

plot_TE_versus_mRNA_uninfected <- ggplot(TE_versus_mRNA_uninfected, aes( x= logFC.mRNA, y = logFC.TE))+
  geom_point()+
  geom_hline(yintercept = 0)+
  geom_vline(xintercept = 0)+
  labs(x= "LogFC mRNA-seq (rep1/rep1)", 
       y = "LogFC Translation efficiency (rep1/rep2)")



```

